pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -1193,6 +1247,83 @@ func (nc *Controller) handleDisruption(zoneToNodeConditions map[string][]*v1.Nod
 	}
 }
 
+func (nc *Controller) podUpdated(oldPod, newPod *v1.Pod) {
+	if newPod == nil {
+		// Nothing to do when pod is deleted.
+		return
+	}
+	if len(newPod.Spec.NodeName) != 0 && (oldPod == nil || newPod.Spec.NodeName != oldPod.Spec.NodeName) {
+		podKey := createPodKey(newPod.Namespace, newPod.Name)
+		nc.podUpdateQueue.Add(podKey)
+	}
+}
+
+func (nc *Controller) doPodProcessingWorker() {
+	for {
+		obj, shutdown := nc.podUpdateQueue.Get()
+		// "podUpdateQueue" will be shutdown when "stopCh" closed;
+		// we do not need to re-check "stopCh" again.
+		if shutdown {
+			return
+		}
+
+		podKey := obj.(string)
+		nc.processPod(podKey)
+	}
+}
+
+func (nc *Controller) processPod(podKey string) {
+	defer nc.podUpdateQueue.Done(podKey)
+	podNamespace, podName := splitPodKey(podKey)
+	pod, err := nc.podLister.Pods(podNamespace).Get(podName)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// If there is no pod, there is no need to requeue.
+			return
+		}
+		utilruntime.HandleError(fmt.Errorf("Failed to read pod %v: %v", podKey, err))
+		nc.podUpdateQueue.Add(podKey)
```

Comment:
Shouldn't be a big issue as errors should be very rare here, so TODO seems fine - there should be exponential backoff on that (or we should use RateLimittedQueue).

Reply:
RateLimittingQueue added.